tee-worker/service/src/main_impl.rs

```diff
@@ -840,7 +805,11 @@ fn spawn_worker_for_shard_polling<InitializationHandler>(
 
 		loop {
 			info!("Polling for worker for shard ({} seconds interval)", POLL_INTERVAL_SECS);
-			if let Ok(Some(_enclave)) = node_api.worker_for_shard(&shard_for_initialized, None) {
+			if let Ok(Some(_account)) = node_api.primary_enclave_identifier_for_shard(
```

Comment:
The `is_primary_enclave` value is calculated before so either we are the primary enclave for shard or there is already other. It sound like we don't need to perform this additional call, what do you think?

Reply:
if `is_primary_enclave` is true, it's all good. If it's false, we don't know if some primary enclave (other than us) exists or no enclave is registered at all.

This line checks if "some" enclave has registered.

This is my understanding purely from the code logic - it might seem redundant because we don't properly take `shard` into consideration, I had comment about it somehwere 😂 

that said I'll keep it for now